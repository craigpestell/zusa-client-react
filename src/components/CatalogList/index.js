/* @flow */

import React from 'react';
import _ from 'lodash';

import { Link } from 'react-router-dom';

import styles from './styles.scss';

type Props = { list: Array<Object> };

export default ({ list }: Props) => (
  <div className={styles.CatalogList}>
    <h4>Catalog List</h4>
    <ul>
      {_.map(list, ({ _id, name }) => (
        <li key={_id}>
          <Link to={`/wo/${_id}`}>{name}</Link>
        </li>
      ))}
    </ul>
  </div>
);
