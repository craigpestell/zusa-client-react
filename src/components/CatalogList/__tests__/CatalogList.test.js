import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import CatalogList from '../index';

describe('<CatalogList />', () => {
  test('renders', () => {
    const mockData = [{ id: '1', name: 'Welly' }];
    const tree = renderer
      .create(
        <MemoryRouter>
          <CatalogList list={mockData} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
