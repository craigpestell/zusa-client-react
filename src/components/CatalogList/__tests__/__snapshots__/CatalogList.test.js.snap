// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatalogList /> renders 1`] = `
<div
  className="CatalogList"
>
  <h4>
    Catalog List
  </h4>
  <ul>
    <li>
      <a
        href="/wo/1"
        onClick={[Function]}
      >
        Welly
      </a>
    </li>
  </ul>
</div>
`;
